metricproviders/datadog/datadog.go

```diff
@@ -165,18 +168,36 @@ func (p *Provider) GarbageCollect(run *v1alpha1.AnalysisRun, metric v1alpha1.Met
 	return nil
 }
 
+func getSecretValue(key string, k8sSecret *v1.Secret, emptyValueOk bool) (string, error) {
+	envKey := strings.ToUpper(strings.ReplaceAll(key, "-", "_"))
+	if value, ok := os.LookupEnv(fmt.Sprintf("DD_%s", envKey)); ok {
+		return value, nil
+	}
+	_, valueMaybe := k8sSecret.Data[key]
+	if emptyValueOk && !valueMaybe {
+		return "", nil
+	}
+
+	return string(k8sSecret.Data[key]), nil
+}
+
 func NewDatadogProvider(logCtx log.Entry, kubeclientset kubernetes.Interface) (*Provider, error) {
 	ns := defaults.Namespace()
 	secret, err := kubeclientset.CoreV1().Secrets(ns).Get(context.TODO(), DatadogTokensSecretName, metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Note that by leaving the logic to fetch the secret here, it does not satisfy your previous benefit:

> This also eliminates the need for granting the argo-rollouts pod the API permission to access the datadog k8s-secret.

Not sure if you care about this as a feature, but I just wanted to call out the fact that it does not really eliminate the RBAC necessary to get the datadog secret.

Reply:
Wow you are right about that. 🤦🏻‍♀️ Thanks for catching that!

Let me address this one more time, properly.

Reply:
Alrighty pushed up 1308004, 3rd time's a charm?